widgetssdk/src/main/java/com/glia/widgets/launcher/EngagementLauncher.kt

```diff
@@ -80,115 +79,142 @@ interface EngagementLauncher {
 
 internal class EngagementLauncherImpl(
     private val activityLauncher: ActivityLauncher,
+    private val configurationManager: ConfigurationManager,
+    private val callVisualizerController: CallVisualizerContract.Controller,
+    private val snackbarController: SnackbarContract.Controller,
     private val hasOngoingSecureConversationUseCase: HasOngoingSecureConversationUseCase,
     private val isQueueingOrLiveEngagementUseCase: IsQueueingOrLiveEngagementUseCase,
-    private val endEngagementUseCase: EndEngagementUseCase,
-    private val configurationManager: ConfigurationManager,
     private val engagementTypeUseCase: EngagementTypeUseCase,
-    private val callVisualizerController: CallVisualizerContract.Controller,
-    private val destroyChatController: () -> Unit,
-    private val destroyCallController: () -> Unit
 ) : EngagementLauncher {
 
-    override fun startChat(context: Context) {
-        if (isQueueingOrLiveEngagementUseCase.isQueueingForMedia) {
-            Logger.i(TAG, "Canceling ongoing queue ticket to create a new one")
-            endEngagementUseCase()
-            destroyChatController()
-        }
+    /**
+     * @return true if there is an ongoing engagement or enqueueing for any media type
+     */
+    private val isEngagementOrQueueing: Boolean
+        get() = isQueueingOrLiveEngagementUseCase.isQueueing || isQueueingOrLiveEngagementUseCase.hasOngoingLiveEngagement
 
+    /**
+     * @return true if there is an ongoing engagement or enqueueing for audio or video media type
+     */
+    private val isMediaEngagementOrMediaQueueing: Boolean
+        get() = engagementTypeUseCase.isMediaEngagement || isQueueingOrLiveEngagementUseCase.isQueueingForMedia
+
+    /**
+     * @return true if there is an ongoing engagement or enqueueing for live chat
+     */
+    private val isChatEngagementOrChatQueueing: Boolean
+        get() = engagementTypeUseCase.isChatEngagement || isQueueingOrLiveEngagementUseCase.isQueueingForLiveChat
+
+    override fun startChat(context: Context) {
         when {
             engagementTypeUseCase.isCallVisualizer -> callVisualizerController.showAlreadyInCvSnackBar()
-            isQueueingOrLiveEngagementUseCase.hasOngoingLiveEngagement || isQueueingOrLiveEngagementUseCase.isQueueingForLiveChat ->
-                activityLauncher.launchChat(context, Intention.RETURN_TO_CHAT)
-
+            isMediaEngagementOrMediaQueueing -> snackbarController.showSnackBar(R.string.entry_widget_call_visualizer_description)
```

Comment:
I think we should have different texts for `ongoing engagement` and `enqueueing` states.
This is true for each function inside `EngagementLauncher`.

Reply:
For the record, PM approved going with the current text.